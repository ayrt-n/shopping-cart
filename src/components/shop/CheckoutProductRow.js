import React from 'react';

function CheckoutProductRow({ product }) {
  const total = (Math.round(product.price * product.quantity * 100) / 100).toFixed(2);

  return (
    <tr>
      <td>{product.name}</td>
      <td>{product.quantity}</td>
      <td>${total}</td>
    </tr>
  );
}

export default CheckoutProductRow;
