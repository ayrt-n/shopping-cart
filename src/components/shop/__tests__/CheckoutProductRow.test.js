import React from 'react';
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import CheckoutProductRow from '../CheckoutProductRow';

describe('CheckoutProductRow component', () => {
  it('renders product details', () => {
    const product = {
      name: 'Test',
      quantity: 5,
      price: 1,
    }

    const { container } = render(<CheckoutProductRow product={product} />)

    expect(container).toMatchSnapshot();
  });
});