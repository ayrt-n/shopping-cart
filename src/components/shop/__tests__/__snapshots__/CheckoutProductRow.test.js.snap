// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutProductRow component renders product details 1`] = `
<div>
  <tr>
    <td>
      Test
    </td>
    <td>
      5
    </td>
    <td>
      $
      5.00
    </td>
  </tr>
</div>
`;
